Model/Filter/FilterBuilder.php

```diff
@@ -0,0 +1,196 @@
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     */
+    public function init(Store $store): void
+    {
+        $this->brand = $this->nostoHelperData->getBrandAttribute($store);
+    }
+
+    /**
+     * @param Item[] $filters
+     * @throws LocalizedException
+     */
+    public function buildFromSelectedFilters($filters)
+    {
+        foreach ($filters as $filter) {
+            $this->mapIncludeFilter($filter);
+        }
+    }
+
+    /**
+     * @param Item $item
+     * @throws LocalizedException
+     */
+    public function mapIncludeFilter(Item $item): void
+    {
+
+        $attributeModel = $item->getFilter()->getData('attribute_model');
+        if($attributeModel === null) {
+            return;
+        }
+
+        /** @var string $frontendInput */
+        $frontendInput = $attributeModel->getData('frontend_input');
+        if ($frontendInput === null) {
+            return;
+        }
+
+        $filterName = $item->getName();
+        $value = '';
+        switch ($frontendInput) {
+            case 'price':
+                $value = $item->getData('value');
+                break;
+            case 'select':
+            case 'multiselect':
+                $value = $item->getData('label');
+                break;
+            case 'date':
+                break;
+            case 'boolean':
+                $value = $item->getData('value') === '1';
+                break;
+            default:
+                $this->logger->debug(sprintf(
+                    'Cannot build include filter for "%s" frontend input type',
+                    $frontendInput
+                ));
+                return;
+        }
+        $this->mapValueToFilter($filterName, $value);
+    }
+
+    /**
+     * @param string $name
+     * @param string|array $value
+     */
+    private function mapValueToFilter(string $name, $value)
+    {
+        if ($this->brand === $name) {
+            $this->includeFilters->setBrands($this->makeArrayFromValue($value));
+            return;
+        }
+
+        switch (strtolower($name)) {
+            case 'price':
+                $this->includeFilters->setPrice(min($value), max($value));
+                break;
+            case 'new':
+                $this->includeFilters->setFresh($value);
+                break;
+            default:
+                $this->includeFilters->setCustomFields($name, $this->makeArrayFromValue($value));
+                break;
+        }
+    }
+
+    /**
+     * @return IncludeFilters
+     */
+    public function getIncludeFilters(): IncludeFilters
+    {
+        return $this->includeFilters;
+    }
+
+    /**
+     * @return ExcludeFilters
+     */
+    public function getExcludeFilters(): ExcludeFilters
+    {
+        return $this->excludeFilters;
+    }
+
+    /**
+     * @param string|array $value
+     * @return array
+     */
+    private function makeArrayFromValue($value)
+    {
+        if (is_string($value)) {
```

Comment:
Should we check `is_numeric` as well? Attributes can have numeric values as well I suppose